bindings/c/test/apitester/TesterMultiTenantCorrectnessWorkload.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+ * TesterMultiTenantCorrectnessWorkload.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "TesterApiWorkload.h"
+#include "TesterUtil.h"
+#include "test/fdb_api.hpp"
+#include <memory>
+#include <map>
+#include <fmt/format.h>
+
+namespace FdbApiTester {
+
+class MultiTenantCorrectnessWorkload : public ApiWorkload {
+public:
+	MultiTenantCorrectnessWorkload(const WorkloadConfig& config) : ApiWorkload(config) {
+		numTenants = config.numTenants;
+	}
+
+private:
+	enum OpType { OP_INSERT, OP_GET, OP_GET_RANGE, OP_CLEAR, OP_CLEAR_RANGE, OP_LAST = OP_CLEAR_RANGE };
+	int numTenants = -1;
+
+	void start() override {
+		schedule([this]() { setupTenants([this]() { runTests(); }, this->numTenants); });
```

Comment:
I would suggest also clearing out all existing tenants at the beginning or at the end of the test. This way we would have an additional test for destroying tenants, and also can reuse the cluster for running multiple tests.

Reply:
This would require us deleting all of the data in the cluster. Is that a reasonable operation at the beginning of the test? Is there an expectation that we leave the cluster in an empty state at the end?

Reply:
The workloads are currently designed to clear out the keyspace they are using before populating new data. This is not a general requirement. It is necessary for the workloads that track all their changes and check if the database reads match the reads from their local store.

In the current tests, workloads do not clear data at the end. If some test fails we cannot guarantee that it has cleared out its data anyway. So it is rather if a workload expects a clean state at the beginning, it should rather clear it out the relevant keyspace by itself.